frame/staking/src/tests.rs

```diff
@@ -3859,6 +3871,78 @@ fn on_finalize_weight_is_nonzero() {
 	})
 }
 
+#[test]
```

Comment:
I don't see any test/example for the most important use case of this PR: 
say we have 8 validators in 3 differnet bags: 
- [a, b, c] [d, e] [f, g, h]

if we iter().take(6) of them, we'd get [a, b, c ,d, e, f]

Reply:
That test is [here](https://github.com/paritytech/substrate/blob/dbfb9512120a4cec144264729649464281d52429/frame/staking/src/voter_bags.rs#L864). I'm open to moving it to the tests file, but I put it in the voter bags module so that I could easily run only voter bags tests as desired.